pkg/snapshot/snapshotter/snapshotter.go

```diff
@@ -736,19 +736,21 @@ func (ssr *Snapshotter) resetFullSnapshotTimer() error {
 	return nil
 }
 
+// checkSnapstoreSecretUpdate checks if the snapstore secret has been updated
 func (ssr *Snapshotter) checkSnapstoreSecretUpdate() bool {
-	ssr.logger.Debug("checking the hash of snapstore secret...")
-	newSnapstoreSecretHash, err := snapstore.GetSnapstoreSecretHash(ssr.snapstoreConfig)
+	ssr.logger.Debug("checking the timestamp of snapstore secret...")
+	newSecretModTime, err := snapstore.GetSnapstoreSecretModTime(ssr.snapstoreConfig)
 	if err != nil {
 		return true
```

Comment:
Please ensure that errors are logged, and additionally, include comments explaining the reasons for returning `true`. This will provide clarity on why the function behaves as it does in specific scenarios.